src/widgets/DataTable/DataTableRow.js

```diff
@@ -88,7 +89,15 @@ const DataTableRow = React.memo(
 
           switch (type) {
             case COLUMN_TYPES.EDITABLE_STRING:
-            case COLUMN_TYPES.EDITABLE_NUMERIC:
+            case COLUMN_TYPES.EDITABLE_NUMERIC: {
+              // Special condition for stocktake counted total quantity cells.
+              // Use the placeholder 'Not counted' when a stocktake item or batch
+              // has not been counted yet.
+              let placeholder = '';
+              if (columnKey === COLUMN_NAMES.COUNTED_TOTAL_QUANTITY) {
+                placeholder = rowData.hasBeenCounted ? '' : tableStrings.not_counted;
+              }
```

Comment:
Maybe can remove the mutable state here, something like:

```
const defaultPlaceholder = '';
const notCountedPlaceholder = tableStrings.not_counted;
const isNotCounted = columnKey === COLUMN_NAMES.COUNTED_TOTAL_QUANTITY && !rowData.hasBeenCounted;
const placeholder = isNotCounted ? notCountedPlaceholder : defaultPlaceholder;

```

Reply:
as much as I don't like mutating - if it makes it easier to read I just do it, and it seems to be easier to read than those lines? (and also more open to other special conditions if we need to use placeholders for other column types?) :(

Reply:
Personally, I find `let` hard to read, because it's harder to reason about the variable state (because it can be changed). Looking at it again, I think I went over the top with variables. You could make the suggestion a bit cleaner:

```
const isNotCounted = columnKey === COLUMN_NAMES.COUNTED_TOTAL_QUANTITY && !rowData.hasBeenCounted;
const placeholder = isNotCounted ? tableStrings.not_counted : '';
```

Let me know either way and I'll merge 👍 .

Reply:
i think no, simply cause the next cell that needs a placeholder will be screwed!